driver-kafka/src/main/java/com/datastax/ebdrivers/kafkaproducer/KafkaStatement.java

```diff
@@ -0,0 +1,141 @@
+
+        // Process key serializer class and schema, if any
+        this.keySerializerClass =
+            stmtDef.getOptionalStringParam("key_serializer_class")
+                .orElse(StringSerializer.class.getName());
+
+        Optional<String> keySerializerSchemaFile =
+            stmtDef.getOptionalStringParam("key_serializer_schema_file");
+
+        if (keySerializerClass.equals("io.confluent.kafka.serializers.KafkaAvroSerializer")
+            && keySerializerSchemaFile.isEmpty() ) {
+            throw new RuntimeException("KafkaAvroSerializer requires key_serializer_schema_file");
+        }
+
+        if (keySerializerSchemaFile.isPresent()) {
+            Path schemaFilePath = Path.of(keySerializerSchemaFile.get());
+            try {
+                this.keySerializerSchema = new AvroSchema(Files.readString(schemaFilePath));
+            } catch (IOException e) {
+                throw new RuntimeException("Error reading key schema file: " + keySerializerSchemaFile, e);
+            }
+        }
+
+        // Process value serializer class and schema, if any
+        this.valueSerializerClass =
+            stmtDef.getOptionalStringParam("value_serializer_class")
+                .orElse(StringSerializer.class.getName());
+
+        Optional<String> valueSerializerSchemaFile =
+            stmtDef.getOptionalStringParam("value_serializer_schema_file");
+
+        if (valueSerializerClass.equals("io.confluent.kafka.serializers.KafkaAvroSerializer")
+            && valueSerializerSchemaFile.isEmpty() ) {
+            throw new RuntimeException("KafkaAvroSerializer requires value_serializer_schema_file");
+        }
+
+        if (valueSerializerSchemaFile.isPresent()) {
+            Path schemaFilePath = Path.of(valueSerializerSchemaFile.get());
+            try {
+                this.valueSerializerSchema = new AvroSchema(Files.readString(schemaFilePath));
+            } catch (IOException e) {
+                throw new RuntimeException("Error reading value schema file: " + valueSerializerSchemaFile, e);
+            }
+        }
+
+        this.topic = stmtDef.getParamOrDefault("topic","default-topic");
+        this.key = stmtDef.getOptionalStringParam("key").orElse("key");
+
+        Properties props = new Properties();
+        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG,  servers);
+        props.put(ProducerConfig.CLIENT_ID_CONFIG, clientId);
+        props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, keySerializerClass);
+        props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, valueSerializerClass);
+        props.put("schema.registry.url", schemaRegistryUrl);
+
+        try {
+            producer = new KafkaProducer<>(props);
+        } catch (Exception e) {
+            logger.error("Error constructing kafka producer", e);
+        }
+    }
+
+    private Object bindKey(long cycle) {
+        Object statement = key;
+        if (keySerializerClass != null &&
```

Comment:
It would be nice to indirect this through an always-defined function at some point.